skytap/resource_skytap_vm.go

```diff
@@ -482,6 +510,36 @@ func resourceSkytapVMUpdate(d *schema.ResourceData, meta interface{}) error {
 	return resourceSkytapVMRead(d, meta)
 }
 
+func buildHardware(d *schema.ResourceData) (*skytap.UpdateHardware, error) {
```

Comment:
Is not this part of the disk update, shall we take out 

```
if v, ok := d.GetOk("disk"); ok && d.HasChange("disk") 
```
to the update and the create a modify method that does the disk change, that will be easy to read, it will follow the standard code convention for resources, and allow us to introduce `partial`


At this stage we need to look at `partial` for this resource modification

Reply:
The create method calls addHardware(). The update method call buildHardware(). The above quoted line is taken from buildHardware(), i.e. the update method. Does this answer your point?
Regarding `partial` isn't this only necessary when multiple network calls are made? The update method only makes one call.

Reply:
buildHardware() is renamed updateHardware() in PR #15

Reply:
I see I thought the update of the hardware operation was done in a different sdk call, it makes sense tho, but my point of taking  the conditional out 

```
if v, ok := d.GetOk("disk"); ok && d.HasChange("disk") 
```

Reply:
Invalid due to code refactor under another PR.